core/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -243,14 +248,35 @@ private void applyDeletedIndices(final ClusterChangedEvent event) {
                 indexSettings = idxService.getIndexSettings();
                 deleteIndex(index, "index no longer part of the metadata");
             } else {
-                final IndexMetaData metaData = previousState.metaData().getIndexSafe(index);
-                indexSettings = new IndexSettings(metaData, settings);
-                indicesService.deleteClosedIndex("closed index no longer part of the metadata", metaData, event.state());
+                IndexMetaData metaData = previousState.metaData().index(index);
+                if (metaData == null) {
+                    // The previous cluster state's metadata also does not contain the index,
+                    // which is what happens on node startup when an index was deleted while the
+                    // node was not part of the cluster.  In this case, try reading the index
+                    // metadata from disk.  If its not there, there is nothing to delete.
+                    try {
+                        metaData = metaStateService.loadIndexState(index);
+                    } catch (IOException e) {
+                        // any other error handling here?
+                        logger.warn("[{}] Deletion request, but unable to read index metadata from disk, " +
+                                    "likely scenario is an index was created and deleted while the node was offline.", e, index);
+                        metaData = null;
+                    }
+                }
+                // index metadata being null means there is nothing to delete
+                if (metaData != null) {
+                    indexSettings = new IndexSettings(metaData, settings);
+                    indicesService.deleteClosedIndex("closed index no longer part of the metadata", metaData, event.state());
+                } else {
+                    indexSettings = null;
+                }
```

Comment:
Only issue with 

`else if (indicesService.canDeleteIndex(Index))`

is that `canDeleteIndexContents` requires the `IndexSettings`, which can't be created until the `IndexMetaData` is loaded, so all that logic will need to go in an `else` block..